src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,544 @@
+            }
+
+            switch (command) {
+            case BYE_COMMAND:
+                return;
+
+            case LIST_COMMAND:
+                runList();
+                break;
+
+            case HELP_COMMAND:
+                printHelp();
+                break;
+
+            case DONE_COMMAND:
+                runDone(input);
+                saveData();
+                break;
+
+            case TODO_COMMAND:
+                runTodo(input);
+                saveData();
+                break;
+
+            case DEADLINE_COMMAND:
+                runDeadline(input);
+                saveData();
+                break;
+
+            case EVENTS_COMMAND:
+                runEvent(input);
+                saveData();
+                break;
+
+            case DELETE_COMMAND:
+                runDeleteCommand(input);
+                saveData();
+                break;
+
+            default:
+                runUnknownCommand(input);
+            }
+        }
+
+    }
+
+
+    /**
+     * FILE IO
+     */
+
+    private static void checkSavedData() {
+        try {
+            readFromFile();
+            System.out.println("Existing records found! Data loaded..." + '\n');
+        } catch (FileNotFoundException e) {
+            System.out.println("No previous records! Starting a new record..." + '\n');
+            createNewFile();
+        }
+    }
+
+    private static void createNewFile() {
+        File file = new File(PATH);
+        try {
+            if (file.createNewFile()) {
+                System.out.println("New file created!" + '\n');
+            }
+        } catch (IOException e) {
+            System.out.println("Error creating new file...");
+            e.printStackTrace();
+        }
+    }
+
+    private static void saveData() {
+        try {
+            writeToFile();
+        } catch (IOException e) {
+            System.out.println("Something went wrong when saving...");
+        }
+    }
+
+    private static void readFromFile() throws FileNotFoundException {
+
+        File dataFile = new File(PATH);
+        Scanner reader = new Scanner(dataFile);
+
+        while (reader.hasNext()) {
+            int lineNumber = 0;
+            String line = reader.nextLine();
+            lineNumber++;
+
+            try {
+                Task task = formTask(line);
+                taskList.add(task);
+                Task.taskCount++;
+            } catch (DataErrorException e) {
+                printDataErrorWarning(lineNumber);
+            }
+        }
+    }
+
+    private static void writeToFile() throws IOException {
+        FileWriter writer = new FileWriter(PATH);
+
+        for (Task t : taskList) {
+            writer.write(formLine(t) + '\n');
+        }
+        writer.close();
+    }
+
+    private static String formLine(Task t) {
+        String line = "";
+
+        if (t instanceof Todo) {
+            line += "T" + DELIM;
+            line += (t.isDone() ? "1" : "0") + DELIM;
```

Comment:
Avoid magic numbers. Consider giving these `1` and `0` a descriptive constant names.